<launch>
    <!-- -loop -->
    <!-- <node pkg="rosbag" type="play" name="player"  args="-loop  $(find gmm_test)/bag/pointcloud_demo.bag"/> -->
    <!-- <node pkg="gmm_test" type="gmm_test" name="gmm" output="screen" />-->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args=" -d $(find gmm_test)/rviz/gmm.rviz" required="true" />-->


    <arg name="agent_id"  default="0"/>
	<arg name="x" default="0"/>
	<arg name="y" default="5"/>
	<arg name="z" default="0"/>

	<arg name="grid_size" default="0.3"/><!-- 体素滤波精度（必须大于等于拟合算法中的精度！！！） -->


	<node pkg="gmm_test" name="GMM_evaluate$(arg agent_id)" type="gmm_evaluate" output="screen">
        <param name="agent_id" value="$(arg agent_id)" type="int"/>
	    <param name="x" value="$(arg x)" type="double"/>
	    <param name="y" value="$(arg y)" type="double"/>
	  	<param name="z" value="$(arg z)" type="double"/>
	  	<param name="grid_size" value="$(arg grid_size)" type="double"/>

		<remap from="/gmm_cloud" to = "/GMM_fitting$(arg agent_id)/gmm_cloud"/>
		<remap from="/gmm_cloud_sample" to = "/GMM_fitting$(arg agent_id)/gmm_cloud_sample"/>


	</node>


</launch>